Sources/CoreCommands/SwiftCommandState.swift

```diff
@@ -1064,25 +1064,33 @@ public final class SwiftCommandState {
         // Try a non-blocking lock first so that we can inform the user about an already running SwiftPM.
         do {
             try workspaceLock.lock(type: .exclusive, blocking: false)
+            let pid = ProcessInfo.processInfo.processIdentifier
+            try String(pid).write(toFile: self.scratchDirectory.appending(".lock").pathString, atomically: true, encoding: .utf8)
```

Comment:
nit: refactor out `self.scratchDirectory.appending(".lock").pathString` in to a variable